src/adc.rs

```diff
@@ -495,11 +493,57 @@ macro_rules! adc_hal {
                         _enabled: PhantomData,
                     }
                 }
+                /// Sets the clock configuration for this ADC. This is common
+                /// between ADC1 and ADC2, so the prec block is used to ensure
+                /// this method can only be called on one of the ADCs (or both,
+                /// using the
+                ///
+                /// Only `CKMODE[1:0]` = 0 is supported
+                fn configure_clock(&mut self, f_adc: Hertz, prec: rec::$Rec, clocks: &CoreClocks) {
+                    let ker_ck = kernel_clk_unwrap(&prec, clocks);
+
+                    // Target mux output. See RM0433 Rev 7 - Figure 136.
+                    #[cfg(feature = "revision_v")]
+                    let f_target = f_adc.raw() * 2;
+
+                    #[cfg(not(feature = "revision_v"))]
+                    let f_target = f_adc.raw();
+
+                    let (divider, presc) = match (ker_ck.raw() + f_target - 1) / f_target {
+                        1 => (1, PRESC_A::Div1),
+                        2 => (2, PRESC_A::Div2),
+                        3..=4 => (4, PRESC_A::Div4),
+                        5..=6 => (6, PRESC_A::Div6),
+                        7..=8 => (8, PRESC_A::Div8),
+                        9..=10 => (10, PRESC_A::Div10),
+                        11..=12 => (12, PRESC_A::Div12),
+                        13..=16 => (16, PRESC_A::Div16),
+                        17..=32 => (32, PRESC_A::Div32),
+                        33..=64 => (64, PRESC_A::Div64),
+                        65..=128 => (128, PRESC_A::Div128),
+                        129..=256 => (256, PRESC_A::Div256),
+                        _ => panic!(),
+                    };
+                    unsafe { &*$ADC_COMMON::ptr() }.ccr.modify(|_, w| w.presc().variant(presc));
+
+                    // Calculate actual value. See RM0433 Rev 7 - Figure 136.
+                    #[cfg(feature = "revision_v")]
+                    let f_adc = Hertz::from_raw(ker_ck.raw() / (divider * 2));
+
+                    // Calculate actual value Revison Y. See RM0433 Rev 7 - Figure 137.
+                    #[cfg(not(feature = "revision_v"))]
+                    let f_adc = Hertz::from_raw(ker_ck.raw() / divider);
+
+                    // Maximum value, for VOS0 only. Other voltage scale values
+                    // result in a lower maximum f_adc
+                    assert!(f_adc.raw() <= 50_000_000);
```

Comment:
I think we want 40 MHz here.

```suggestion
                    assert!(f_adc.raw() <= 40_000_000);
```

c.f. https://github.com/quartiq/stabilizer/pull/584#discussion_r945547973